src/Security/Items.cs

```diff
@@ -237,11 +234,14 @@ internal SecKeyChain (NativeHandle handle)
 				copy.LowlevelSetObject (CFBoolean.TrueHandle, SecItem.ReturnData);
 				var n = SetLimit (copy, max);
 				
-				IntPtr ptr;
-				result = SecItem.SecItemCopyMatching (copy.Handle, out ptr);
+				result = SecItem.SecItemCopyMatching (copy.Handle, out IntPtr ptr);
 				n = null;
 				if (result == SecStatusCode.Success)
-					return NSArray.ArrayFromHandleFunc<SecRecord> (ptr, (element) => new SecRecord (Runtime.GetNSObject<NSMutableDictionary> (element, false)!))!;
+					return CFArray.ArrayFromHandleFunc<SecRecord> (ptr, (element) => {
+						CFObject.CFRetain (element);
+						var dictionary = Runtime.GetNSObject<NSMutableDictionary> (element, true)!;
```

Comment:
So, from my understanding, the container owns the elements, so we need to retain them before we grab the element, otherwise the container going away means the element will go away as well.